### PR TITLE
Update Edit FI Naming

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

- Update `BTPayPalEditRequest` to `BTPayPalVaultEditRequest`
    - Update `token` to `encryptedBillingAgreementID`
- Add new `BTPayPalVaultEditResult`
- Update `tokenize` method to `edit`
- Update tests

### Checklist

- [x] Added a changelog entry

### Authors

- @jaxdesmarais

### CODE REVIEW

**Comment by jaxdesmarais** [comment]
@scannillo and @tdchow want to make sure I get your eyes on the finalized naming in this PR

File: Sources/BraintreePayPal/BTPayPalVaultEditResult.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+/// A result of the Edit FI flow used to display a customers updated payment details in your UI
+/// - Warning: This feature is currently in beta and may change or be removed in future releases.
+public class BTPayPalVaultEditResult {
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/BTPayPalVaultEditResult.swift:5]: Same here, could maybe drop `Vault` and use this result for checkout and vault edit FI flows in the future.

**Review by scannillo — APPROVED** [review]

File: Sources/BraintreePayPal/BTPayPalVaultEditRequest.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+/// - Warning: This feature is currently in beta and may change or be removed in future releases.
+public class BTPayPalVaultEditRequest {
+
+    private let encryptedBillingAgreementID: String
+
+    /// Initializes a PayPal Edit Request for the Edit FI flow
+    /// - Parameters:
+    ///   - encryptedBillingAgreementID: Required: Used to initiate tokenize call to edit funding instrument in customer's PayPal account.
+    //   TODO: specify endpoint for merchant to retrieve the token
+    public init(encryptedBillingAgreementID: String) {
```
Comment (scannillo) [inline Sources/BraintreePayPal/BTPayPalVaultEditRequest.swift:13]: I think you saw the thread started re: what this name should be across server & client SDKs. It _may_ change.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 * BraintreePayPal
   * Add PayPal edit funding instrument flow (BETA)
     * Add `BTPayPalEditRequest` for edit FI flow
-    * Add `BTPayPalClient.tokenize(_:completion:)` method that takes in a `BTPayPalEditRequest`
+    * Add `BTPayPalClient.edit(_:completion:)` method that takes in a `BTPayPalEditRequest`
```
Comment (agedd) [inline CHANGELOG.md:7]: nit: take or leave it 
```suggestion
    * Add `BTPayPalClient.edit(_:completion:)` method that takes in a `BTPayPalVaultEditResult`
```

File: Sources/BraintreePayPal/BTPayPalClient.swift [hunk 4]
```diff
@@ -189,12 +189,12 @@ import BraintreeDataCollector
     /// If the user cancels out of the flow, the error code will be `.canceled`.
```
Comment (tdchow) [inline Sources/BraintreePayPal/BTPayPalClient.swift:189]:
```    
    /// Tokenize a PayPal request to be used with the Edit FI flow.
    ///
    /// On success, you will receive an instance of `BTPayPalAccountNonce`; on failure or user cancelation you will receive an error.
    /// If the user cancels out of the flow, the error code will be `.canceled`.```
    
We may want to update this comment, removing the references to tokenize and nonce.

**Review by agedd — APPROVED** [review]

File: Sources/BraintreePayPal/BTPayPalClient.swift [hunk 5]
```diff
@@ -189,12 +189,12 @@ import BraintreeDataCollector
...
     ///
     /// - Parameters:
-    ///   - request: A `BTPayPalEditRequest`
+    ///   - request: A `BTPayPalVaultEditRequest`
     ///   - completion: This completion will be invoked exactly once when tokenization is complete or an error occurs.
```
Comment (tdchow) [inline Sources/BraintreePayPal/BTPayPalClient.swift:193]:
```suggestion
    ///   - completion: This completion will be invoked exactly once when the edit flow is complete or an error occurs.
```

File: Sources/BraintreePayPal/BTPayPalClient.swift [hunk 6]
```diff
@@ -205,17 +205,17 @@ import BraintreeDataCollector
     /// On success, you will receive an instance of `BTPayPalAccountNonce`; on failure or user cancelation you will receive an error.
     /// If the user cancels out of the flow, the error code will be `.canceled`.
     ///
-    /// - Parameter request: A `BTPayPalEditRequest`
+    /// - Parameter request: A `BTPayPalVaultEditRequest`
     /// - Returns: A `BTPayPalAccountNonce` if successful
```
Comment (tdchow) [inline Sources/BraintreePayPal/BTPayPalClient.swift:209]:
```suggestion
    /// - Returns: A `BTPayPalVaultEditResult` if successful
```

File: Sources/BraintreePayPal/BTPayPalVaultEditRequest.swift (hunk 2 above)
Comment (jaxdesmarais) [reply Sources/BraintreePayPal/BTPayPalVaultEditRequest.swift:13]: Yeah, will mark this as "do not merge" until that's decided

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,7 +4,7 @@
 * BraintreePayPal
   * Add PayPal edit funding instrument flow (BETA)
     * Add `BTPayPalEditRequest` for edit FI flow
```
Comment (jaxdesmarais) [inline CHANGELOG.md:6]:
```suggestion
    * Add `BTPayPalVaultEditRequest` for edit FI flow
```

File: Sources/BraintreePayPal/BTPayPalClient.swift [hunk 8]
```diff
@@ -205,17 +205,17 @@ import BraintreeDataCollector
     /// On success, you will receive an instance of `BTPayPalAccountNonce`; on failure or user cancelation you will receive an error.
```
Comment (tdchow) [inline Sources/BraintreePayPal/BTPayPalClient.swift:205]: Same doc changes here.

File: Sources/BraintreePayPal/BTPayPalClient.swift (hunk 4 above)
Comment (jaxdesmarais) [reply Sources/BraintreePayPal/BTPayPalClient.swift:189]: Good catch, think it caught them all here: https://github.com/braintree/braintree_ios/pull/1369/commits/a51c73d8925b0e874bdc2d1f191ec4f1ba742169

File: UnitTests/BraintreePayPalTests/BTPayPalClient_Tests.swift [hunk 9]
```diff
@@ -127,19 +127,19 @@ class BTPayPalClient_Tests: XCTestCase {
     }
 
     func testTokenizePayPalEditRequest_whenResponseIsSuccessful_returnsPayPalAccountNonce() {
```
Comment (KunJeongPark) [inline UnitTests/BraintreePayPalTests/BTPayPalClient_Tests.swift:129]: We probably need to change this return type.

**Review by warmkesselj — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 10]
```diff
@@ -6,8 +6,8 @@
   * Add error code and error message for `exceededTimeoutLimit`  
 * BraintreePayPal
   * Add PayPal edit funding instrument flow (BETA)
-    * Add `BTPayPalEditRequest` for edit FI flow
-    * Add `BTPayPalClient.tokenize(_:completion:)` method that takes in a `BTPayPalEditRequest`
+    * Add `BTPayPalVaultEditRequest` for edit FI flow
+    * Add `BTPayPalClient.edit(_:completion:)` method that takes in a `BTPayPalVaultEditResult`
```
Comment (jaxdesmarais) [inline CHANGELOG.md:10]:
```suggestion
    * Add `BTPayPalClient.edit(_:completion:)` method that takes in a `BTPayPalVaultEditRequest`
```

**Pull request merged**